fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/catalog/FlinkCatalog.java

```diff
@@ -371,8 +379,42 @@ public void alterTable(ObjectPath objectPath, CatalogBaseTable catalogBaseTable,
     @Override
     public List<CatalogPartitionSpec> listPartitions(ObjectPath objectPath)
             throws TableNotExistException, TableNotPartitionedException, CatalogException {
-        // TODO: use admin.listPartitionInfos()
-        return Collections.emptyList();
+        // TODO lake table should support.
+        if (objectPath.toString().contains("$lake")) {
+            return Collections.emptyList();
+        }
+
+        try {
+            TablePath tablePath = toTablePath(objectPath);
+            TableInfo tableInfo = admin.getTableInfo(tablePath).get();
+            List<String> partitionKeys = tableInfo.getPartitionKeys();
+
+            List<PartitionInfo> partitionInfos = admin.listPartitionInfos(tablePath).get();
+            List<CatalogPartitionSpec> catalogPartitionSpecs = new ArrayList<>();
+            for (PartitionInfo partitionInfo : partitionInfos) {
+                String[] partitionSpec = partitionInfo.getPartitionName().split("\\$");
+                checkArgument(partitionKeys.size() == partitionSpec.length);
+                Map<String, String> flinkPartitionSpec = new HashMap<>();
+                for (int i = 0; i < partitionKeys.size(); i++) {
+                    flinkPartitionSpec.put(partitionKeys.get(i), partitionSpec[i]);
+                }
+                catalogPartitionSpecs.add(new CatalogPartitionSpec(flinkPartitionSpec));
+            }
+            return catalogPartitionSpecs;
+        } catch (Exception e) {
+            Throwable t = ExceptionUtils.stripExecutionException(e);
+            if (isTableNotExist(t)) {
+                throw new TableNotExistException(getName(), objectPath);
+            } else if (isTableNotPartitioned(t)) {
+                throw new TableNotPartitionedException(getName(), objectPath);
+            } else {
+                throw new CatalogException(
+                        String.format(
+                                "Failed to list partitions of table %s in %s",
+                                objectPath, getName()),
+                        t);
+            }
```

Comment:
Cover the exceptions in `FlinkCatalogTest`

Reply:
Done!